azure/service.ts

```diff
@@ -21,23 +21,241 @@ import { RunError } from "@pulumi/pulumi/errors";
 import * as shared from "./shared";
 
 import * as docker from "@pulumi/docker";
-import * as config from "./config";
+import * as utils from "./utils";
 
 export class Service extends pulumi.ComponentResource implements cloud.Service {
     public readonly name: string;
+    public readonly groups: azure.containerservice.Group[];
 
     public readonly endpoints: pulumi.Output<cloud.Endpoints>;
     public readonly defaultEndpoint: pulumi.Output<cloud.Endpoint>;
 
     public readonly getEndpoint: (containerName?: string, containerPort?: number) => Promise<cloud.Endpoint>;
 
     constructor(name: string, args: cloud.ServiceArguments, opts?: pulumi.ResourceOptions) {
-        super("cloud:service:Service", name, {}, opts);
+        let containers: cloud.Containers;
```

Comment:
>  Any thoughts on how to structure things to avoid some of this?

Yes, i think we should have a 'common' layer where we put this.  I would *prefer* to keep this out of 'api' as i would like that to really just be the interfaces.  I'm def happy to explore this as part of this work.

Reply:
Opened https://github.com/pulumi/pulumi-cloud/issues/565 to track this.